pkg/cmd/pr/close/close.go

```diff
@@ -153,7 +153,7 @@ func closeRun(opts *CloseOptions) error {
 				return nil
 			}
 		} else {
-			if err := api.BranchDeleteRemote(apiClient, baseRepo, pr.HeadRefName); err != nil {
```

Comment:
Following up on an internal conversation with @samcoe about these changes, he suggested a novel solution to this problem effectively reducing much of the code changes in the PR:

add the [`-r` flag](https://git-scm.com/docs/git-branch#Documentation/git-branch.txt--r) to the underlying git client code used to delete the local branch

https://github.com/cli/cli/blob/785a340f78ac5201f4652f6eb08929f0b2ef7621/git/client.go#L338-L349

@asakatida : what are you thoughts?

Reply:
I'm curious how that is going to interact with edge cases. Never used the flag so I worry about testing it.

- if there are multiple remotes with the matching branch name, does that flag handle deleting from only the one remote that the PR is linked to?
- does this delete both the remote and local branch in one command?

I can investigate and figure it out if needed. Less code is definitely better

Reply:
if it does leverage git config for tracking branch when picking one remote, then it would definitely be better than my code defaulting to origin

Reply:
Also, the one other nice thing about the existing code is it works outside a git repo. The new version would only delete remote when run from a local repo

Reply:
the command sequence turns out to be `git branch -D feature` + `git branch -d -r origin/feature`. Nothing about the tracking branch is queried, and it requires 2 commands, same as the amount of code here.

Reply:
Summarizing my thoughts above. As compared to current PR code, there is no additional edge case covered by the `-r` flag, and no difference in the lines of code to achieve the results.

was there something else in the private discussion that I failed to test above? @samcoe or @andyfeller

Reply:
@asakatida I think the big advantage of using the `-r` flag is that we can keep the code that modifies the local git repo in a singular place so it is easy to reason about and easier to maintain in the future. Basically we are not mixing API calls and `git` calls depending on if the user is in a local git repository. To me that is enough of a reason to go that route. 

Additionally, I do think it would result in less code changes as we would just need to modify [here](https://github.com/cli/cli/blob/b62e977c91af1fc2322b7d0e1addc61d192bc8e2/git/client.go#L338) to run another `git` command.

Reply:
okay, I follow what you want now. It's been updated